src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -313,21 +324,19 @@ updateShellAndSC(unsigned int  boardIdx, DSAInfo& candidate, bool& reboot)
   bool same_dsa = false;
   bool same_bmc = false;
   DSAInfo current = flasher.getOnBoardDSA();
-  if (!current.name.empty()) {
-    same_dsa = (candidate.name == current.name &&
-      candidate.matchId(current));
-
-    // Always update Arista devices
-    if (candidate.vendor_id == ARISTA_ID)
-        same_dsa = false;
-
-    // getOnBoardDSA() returns an empty bmcVer in the case there is no SC,
-    // so do not update
-    if (current.bmcVer.empty())
-        same_bmc = true;
-    else
-        same_bmc = (candidate.bmcVer == current.bmcVer);
-  }
+  isSameShellOrSC(candidate, current, &same_dsa, &same_bmc);
+
+  // Always update Arista devices
+  if (candidate.vendor_id == ARISTA_ID)
+    same_dsa = false;
+
+  // getOnBoardDSA() returns an empty bmcVer in the case there is no SC,
+  // so do not update
+  if (current.bmcVer.empty())
```

Comment:
```suggestion
  same_dsa = current.bmcVer.empty() ? true : (candidate.bmcVer == current.bmcVer);
```

Reply:
Going forward, same_bmc will have to support more conditions. If I merge the if statement in this, it will be a very long logic in one line. We can revisit this when I have added the new logic.

Reply:
Understand.  There are trade offs between the code we implement today and adding support for tomorrow.  

For classes, libraries, etc. it makes sense to architect the code base to lead the developers to correctly integrate new functionality.  With regards to methods, these are small enough that if needed it can be refactored completely with new functionality as needed.   

In this case, it would be best to focus the code for ease-of-reading and let the next developer determine how best to add additional conditionals.   In this case the conditional "?" would be a better option.